src/inc-js-variants.php

```diff
@@ -515,44 +456,47 @@ function lovd_highlightInput (oElement)
+            }).change();
+        }
+    });
+
+    // We disable the effect of the Enter key, since we do not want the user
+    //  to be able to easily submit the form (which is the default effect of Enter),
+    //  since this would mean the onchange would be bypassed and the HGVS check would
+    //  not be run.
+    $(document).on("keydown", "input[name*='DNA']", function(event) {
+        return event.key != "Enter";
+    });
 
     if (oTranscriptVariants[0] != undefined) {
         // Add the buttons and images at the end of the transcripts DNA fields.
-        oTranscriptVariants.parent().append('&nbsp;&nbsp;<IMG style="display:none;" align="top" width="16" height="16">&nbsp;<BUTTON class="mapVariant" type="button" onclick="lovd_convertPosition(this); return false;" style="display:none;">Map to genome</BUTTON>');
+        oTranscriptVariants.parent().append('&nbsp;&nbsp;<IMG src="gfx/trans.png" align="top" width="16" height="16">&nbsp;');
         var nTranscriptVariants = oTranscriptVariants.size();
         for (i=0; i < nTranscriptVariants; i++) {
-            // Add an artificial attribute "id_ncbi" to the transcripts DNA input field. This is needed to link the response from Mutalyzer to this field, if needed.
-            $(oTranscriptVariants[i]).attr('id_ncbi', aTranscripts[$(oTranscriptVariants[i]).attr('name').substring(0, <?php echo $_SETT['objectid_length']['transcripts']; ?>)][0]);
+            // Add an artificial attribute "id_ncbi" to the transcripts DNA input field. This is needed to link the response from VariantValidator to this field.
+            $(oTranscriptVariants[i]).data('id_ncbi', aTranscripts[$(oTranscriptVariants[i]).attr('name').substring(0, <?php echo $_SETT['objectid_length']['transcripts']; ?>)][0]);
         }
         // Add an onChange event that runs lovd_checkHGVS.
         oTranscriptVariants.change(lovd_checkHGVS);
-        // Add same function to the onKeyUp event, but then it will check itself if the variant is likely to be complete.
-        oTranscriptVariants.keyup(lovd_checkHGVS);
 
         var oProteinVariants = $('#variantForm input[name$="_VariantOnTranscript/Protein"]');
         if (oProteinVariants[0] != undefined) {
             // Add the buttons and images at the end of the protein description fields.
-            oProteinVariants.parent().append('&nbsp;&nbsp;<IMG src="gfx/trans.png" style="display:inline;" align="top" width="16" height="16">&nbsp;<BUTTON class="proteinChange" type="button" onclick="lovd_getProteinChange(this); return false;" style="display:none;">Predict</BUTTON>');
-        }
-
-        if (oTranscriptVariants.val() !== '') {
-            // Variant field already has content, check HGVS now because if we're on an edit form we
-            // want the buttons to be ready.
-            oTranscriptVariants.change();
+            oProteinVariants.parent().append('&nbsp;&nbsp;<IMG src="gfx/trans.png" style="display:none;" align="top" width="16" height="16">&nbsp;');
```

Comment:
Ah, here I see one `"display:none"` that is removed everywhere else. If we remove this style, we can also get rid of all the `.show()` calls out there.